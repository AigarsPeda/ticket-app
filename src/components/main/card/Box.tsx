import React from "react";

interface Props {
  cardValue: string | number;
  title: string;
  iconClass: string;
  cardValueClass: string;
}

const Box: React.FC<Props> = (props) => {
  const { cardValue, title, iconClass, cardValueClass } = props;
  return (
    <div className="col-md-6 col-xl-4">
      <div className="card-box border">
        <i className={`${iconClass} font-24`}></i>
        <h3 className={cardValueClass}>{cardValue}</h3>
        <p className="text-uppercase font-13 font-weight-medium mb-1">
          {title}
        </p>
      </div>
    </div>
  );
};

export default Box;
