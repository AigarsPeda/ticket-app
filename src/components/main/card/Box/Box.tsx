import React from "react";
import { useHistory } from "react-router-dom";

interface Props {
  cardValue: string | number;
  title: string;
  iconClass: string;
  cardValueClass?: string;
  status: string;
  type: string;
}

const Box: React.FC<Props> = (props) => {
  const { cardValue, title, iconClass, cardValueClass, status, type } = props;
  const history = useHistory();

  const navigateToPage = (status: any, type: any) => {
    history.push(`/tickets/${status}/${type}`);
  };

  return (
    <div
      className="col-md-6 col-xl-4"
      onClick={() => navigateToPage(status, type)}
    >
      <div className="card-box border">
        <i className={`${iconClass} font-24`}></i>
        <h3 className={cardValueClass}>{cardValue}</h3>
        <p className="text-uppercase font-13 font-weight-medium mb-1">
          {title}
        </p>
      </div>
    </div>
  );
};

export default Box;
