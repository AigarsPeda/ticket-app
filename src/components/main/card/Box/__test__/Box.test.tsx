import React from "react";
import ReactDOM from "react-dom";

// import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/matchers";

import renderer from "react-test-renderer";
import Box from "../Box";
import Adapter from "enzyme-adapter-react-16";
import Enzyme, { mount } from "enzyme";

Enzyme.configure({ adapter: new Adapter() });

describe("<Box /> component", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <Box title="Total Tickets" cardValue={2} iconClass="fas fa-tag" />,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("testing table", () => {
    const wrapper = mount(
      <Box title="Total Tickets" cardValue={2} iconClass="fas fa-tag" />
    );
    // console.log(wrapper.debug());

    expect(wrapper.find("p").text()).toBe("Total Tickets");
  });

  it("matches snapshot", () => {
    const tree = renderer
      .create(
        <Box title="Total Tickets" cardValue={2} iconClass="fas fa-tag" />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
