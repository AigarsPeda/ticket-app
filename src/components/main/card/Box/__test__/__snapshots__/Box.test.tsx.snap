// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Box /> component matches snapshot 1`] = `
<div
  className="col-md-6 col-xl-4"
>
  <div
    className="card-box border"
  >
    <i
      className="fas fa-tag font-24"
    />
    <h3>
      2
    </h3>
    <p
      className="text-uppercase font-13 font-weight-medium mb-1"
    >
      Total Tickets
    </p>
  </div>
</div>
`;
